hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheRecordStore.java

```diff
@@ -522,14 +522,14 @@ protected void publishBatchedEvents(String cacheName, CacheEventType cacheEventT
         }
     }
 
-    protected boolean compare(Object v1, Object v2) {
+    protected boolean isEqual(Object v1, Object v2) {
         if (v1 == null && v2 == null) {
             return true;
         }
         if (v1 == null || v2 == null) {
             return false;
         }
-        return v1.equals(v2);
+        return toStorageValue(v1).equals(toStorageValue(v2));
```

Comment:
The method is just used in this class. All usages called `toStorageValue()` on the values. This is an optimization to save these calls, if at least one of the values is `null`. The renaming makes more clear, what the result of the comparison is (`compare()` methods normally return -1, 0 and 1).